pkg/api/meta/help_test.go

```diff
@@ -114,7 +115,10 @@ type fakePtrInterfaceList struct {
 	Items *[]runtime.Object
 }
 
-func (f fakePtrInterfaceList) IsAnAPIObject() {}
+func (obj fakePtrInterfaceList) SetGroupVersionKind(gvk *unversioned.GroupVersionKind) {}
```

Comment:
I'm going to punt on this refactor until the rest of the refactors are
done, if you don't mind.  We can do a sweep at that point.

On Thu, Dec 10, 2015 at 7:53 AM, David Eads <notifications@github.com>
wrote:

> In pkg/api/meta/help_test.go
> <https://github.com/kubernetes/kubernetes/pull/18473#discussion_r47222164>
> :
>
> > @@ -114,7 +115,10 @@ type fakePtrInterfaceList struct {
> >  	Items *[]runtime.Object
> >  }
> >
> > -func (f fakePtrInterfaceList) IsAnAPIObject() {}
> > +func (obj fakePtrInterfaceList) SetGroupVersionKind(gvk *unversioned.GroupVersionKind) {}
>
> There's been at least one complaint about gvk and defending the choice
> isn't easy to do.
>
> @derekwaynecarr <https://github.com/derekwaynecarr> wrote a treatise on
> names #18379 (comment)
> <https://github.com/kubernetes/kubernetes/pull/18379#discussion_r47100949>
> and on reflection, I do like kind better.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/18473/files#r47222164>.
>